subhub/hub/stripe/customer.py

```diff
@@ -254,3 +293,20 @@ def run(self):
             raise ClientError(
                 f"userid is None for customer {self.payload.object.customer}"
             )
+
+    def get_latest_invoice(self, invoice_id):
+        latest_invoice = stripe.Invoice.retrieve(id=invoice_id)
+        logger.info("get latest invoice", latest_invoice=latest_invoice)
+        if latest_invoice.charge is None or latest_invoice.number is None:
+            self.get_latest_invoice(invoice_id)
+        return latest_invoice
+
+    def get_latest_charge(self, charge):
+        latest_charge = stripe.Charge.retrieve(id=charge)
+        logger.info("get latest charge", latest_charge=latest_charge)
+        if (
+            latest_charge.payment_method_details.card.last4 is None
+            or latest_charge.payment_method_details.card.brand is None
+        ):
+            self.get_latest_charge(charge)
```

Comment:
So the algorithm is that if for a payment type of credit card, if the last 4 digits of that payment type is not available or the payment credit card brand (VISA, Mastercard, etc) is not available, recursively call with the charge passed in?  I might be reading this incorrectly given the lack of sleep right now and all....

Reply:
Yes, the idea is if Stripe hasn't updated value prior to making this call (has happened already) then call it again till the value is updated so we can ship it to Basket.

Reply:
So question on that algorithm then, how long are you willing to poll their API and should we have any wait period in between calls to avoid getting throttled from excessive calls?

Reply:
Do you recommend doing a time.sleep(0.2) before calling again??  Something else?